haystack/components/preprocessors/document_cleaner.py

```diff
@@ -113,9 +113,13 @@ def _remove_empty_lines(self, text: str) -> str:
         :param text: Text to clean.
         :returns: The text without empty lines.
         """
-        lines = text.split("\n")
-        non_empty_lines = filter(lambda line: line.strip() != "", lines)
-        return "\n".join(non_empty_lines)
+        pages = text.split("\f")
+        lines_per_page = [page.split("\n") for page in pages]
+        non_empty_lines = []
+        for lines in lines_per_page:
+            non_empty_lines.append(filter(lambda line: line.strip() != "", lines))
+        lines_without_empty_lines = ["\n".join(lines) for lines in non_empty_lines]
+        return "\f".join(lines_without_empty_lines)
```

Comment:
We could simplify this too
```python
def _remove_empty_lines(self, text: str) -> str:
    return "\f".join(
        "\n".join(line for line in page.split("\n") if line.strip())
        for page in text.split("\f")
    )
```
better to
```python
def _remove_empty_lines(text: str) -> str:
    pages = text.split("\f")
    cleaned_pages = ["\n".join(line for line in page.split("\n") if line.strip()) for page in pages]
    return "\f".join(cleaned_pages)
```
The second version is also more consistent with the other methods then. The docstring can remain as before. In my opinion it's still readable. What do you think @vblagoje ?

Reply:
Yes, second version is easier to debug as well.